doc/nrf/ug_pelion.rst

```diff
@@ -0,0 +1,104 @@
+* :option:`CONFIG_PELION_UPDATE` - enables device firmware update (DFU) feature.
+* :option:`CONFIG_PELION_TRANSPORT_MODE_TCP`, :option:`CONFIG_PELION_TRANSPORT_MODE_UDP`, :option:`CONFIG_PELION_TRANSPORT_MODE_UDP_QUEUE`- choice selecting the transport protocol used by the library.
+
+For more information and a full list of the library configuration options, read the `Zephyr integration tutorial`_ in the Pelion documentation.
+
+Mbed TLS
+========
+
+The Pelion Device Management library is depending on a properly configured Mbed TLS library.
+To simplify the development within the |NCS|, a predefined set of Mbed TLS configuration options was prepared.
+Set compatible with :ref:`nrfxlib:nrf_security` can be enabled using :option:`CONFIG_PELION_NRF_SECURITY` configuration option.
+
+Make sure to properly configure memory region used by Mbed TLS library for a dynamic allocations.
+Memory for this library is allocated from either the libc heap or a dedicated buffer, depending on a configuration.
+
+Device provisioning
+===================
+
+You must provision the device before it can connect with the Pelion Device Management Portal.
+
+When using Pelion, you can complete the provisioning process using either a production tool (Factory Configurator Utility for factory provisioning) or the developer flow process.
+For more information about the provisioning process, see `Provisioning devices for Pelion Device Management`_ in the Pelion documentation.
+
+For details about provisioning the device with the update resources, see the `Device Management Update`_ guide in the Pelion documentation.
+
+.. note::
+   Zephyr's port of Pelion might not support the most recent update client features.
+   The Device Management Update link points to the documentation of the version compatible with the |NCS|.
+   For details, refer to the `Pelion Device Management documentation`_ and `Pelion Device Management release notes`_.
+
+Flash memory layout
+===================
+
+The Pelion Device Management library requires that a partition named ``pelion_storage`` is defined on flash.
+The partition is used to store the non-volatile data like credentials and identifiers.
+Since data kept are fragile it is recommended to keep the partition on the SoC internal flash or otherwise secure access to the memory.
+
+Dynamic allocation memory
+=========================
+
+The Pelion Device Management library depends on the new libc.
+In case of targets without the memory management unit (MMU), given that the application is not running in the Userspace, the portion of RAM that remains unallocated for any other purpose is used by the libc heap.
+Make sure to leave sufficient space for heap so the Pelion Device Management library may work correctly.
+
+Network backend
+===============
+
+The Pelion Device Management library uses standard POSIX sockets for the network communication.
+It was tested with the following network backends:
+* LTE using the :ref:`lte_lc_readme`
```

Comment:
Discussed f2f: The link provided here is not optimal. We either need to find a better one or expand the docs.

Reply:
expanded somehow